### PR TITLE
ENH: simplify usage of simple_fixture with skipif-wrapped args

### LINKED ISSUE #1149: Fix simple_fixture and never_skip compatibility
These are currently somewhat in-compatible and code like this is needed:

```python
fft_impl = simple_fixture('fft_impl',
                          [odl.util.testutils.never_skip('numpy'),
                           odl.util.testutils.skip_if_no_pyfftw('pyfftw')],
                          fmt=" {name} = '{value.args[1]}' ")
```

### PR DESCRIPTION
**Pull request by kohr-h**
Closes #1149

### CODE REVIEW

File: odl/test/largescale/tomo/ray_transform_slow_test.py [hunk 1]
```diff
@@ -49,7 +49,7 @@
               skip_if_no_astra_cuda('helical astra_cuda uniform'),
               skip_if_no_skimage('par2d skimage uniform')]
 
-projector_ids = ['geom={}, impl={}, angles={}'
+projector_ids = [" geom = {}, impl = '{}', angles={} "
```
Comment (adler-j) [inline odl/test/largescale/tomo/ray_transform_slow_test.py]: Perhaps we should add something like this to simple_fixture

File: odl/test/largescale/trafos/fourier_slow_test.py [hunk 2]
```diff
@@ -27,33 +27,20 @@
...
               odl.uniform_discr([-2, -2, -2], [2, 2, 2], [200, 200, 200],
                                 dtype='complex')]
 
-dom_ids = [' {!r} '.format(dom) for dom in dom_params]
-
-
-@pytest.fixture(scope="module", ids=dom_ids, params=dom_params)
-def domain(request):
-    return request.param
+domain = simple_fixture('domain', dom_params)
```
Comment (adler-j) [inline odl/test/largescale/trafos/fourier_slow_test.py]: Move dom_params to inline here.

Do this "across the board"

File: odl/util/testutils.py [hunk 3]
```diff
@@ -236,29 +236,53 @@ def simple_fixture(name, params, fmt=None):
...
-                # String type
-                fmt = " {name} = '{value}' "
-
-        ids = [fmt.format(name=name, value=value) for value in params]
+            fmt_str = " {name} = '{value}' "
+            fmt_default = " {name} = {value} "
+
+            ids = []
+            for p in params:
+                # TODO: other types of decorators?
```
Comment (adler-j) [inline odl/util/testutils.py]: Nice Todo, perhaps make an issue and reference it here

File: odl/util/testutils.py [hunk 4]
```diff
@@ -236,29 +236,53 @@ def simple_fixture(name, params, fmt=None):
...
+                        ids.append(fmt_default.format(name=name,
+                                                      value=p.args[1]))
+                    else:
+                        ids.append(fmt_str.format(name=name,
+                                                  value=p.args[1]))
+                else:
+                    try:
+                        p + ''
+                    except TypeError:
+                        ids.append(fmt_default.format(name=name,
```
Comment (adler-j) [inline odl/util/testutils.py]: Rather newline after append than the comma here

File: odl/util/testutils.py [hunk 5]
```diff
@@ -236,29 +236,53 @@ def simple_fixture(name, params, fmt=None):
...
+            fmt_str = " {name} = '{value}' "
+            fmt_default = " {name} = {value} "
+
+            ids = []
+            for p in params:
+                # TODO: other types of decorators?
+                if (isinstance(p, _pytest.mark.MarkDecorator) and
+                        p.name == 'skipif'):
+                    # Unwrap the wrapped object in the decorator
+                    try:
```
Comment (adler-j) [inline odl/util/testutils.py]: Why not change this whole complicated logic to "a nice equivalent of"

If is strong(p):
    value = "'{}'".format(p)
else:
    value = p

File: odl/test/largescale/trafos/fourier_slow_test.py [hunk 6]
```diff
@@ -27,33 +27,20 @@
...
 
 
-impl_params = [never_skip('numpy'), skip_if_no_pyfftw('pyfftw')]
-impl_ids = [" impl = '{}'".format(p.args[1]) for p in impl_params]
-
-
 # bug in pytest (ignores pytestmark) forces us to do this this
+impl_params = [never_skip('numpy'), skip_if_no_pyfftw('pyfftw')]
 largescale = " or not pytest.config.getoption('--largescale')"
 impl_params = [pytest.mark.skipif(p.args[0] + largescale, p.args[1])
```
[the remark this answers is not in the document]
Comment (kohr-h) [reply odl/test/largescale/trafos/fourier_slow_test.py]: Moving the list comprehension into the function argument.

File: odl/util/testutils.py (hunk 3 above)
Comment (kohr-h) [reply odl/util/testutils.py]: Not sure if it's really a TODO, therefore the question mark. Depends on whether we ever encounter another type of decorator that we may want to unwrap.

File: odl/util/testutils.py (hunk 5 above)
Comment (kohr-h) [reply odl/util/testutils.py]: Like, making this a utility function?
```python
def is_string(obj):
    try:
        obj + ''
    except TypeError:
        return False
    else:
        return True
```

Comment (adler-j) [reply odl/util/testutils.py]: Oh God yes please!

But also the second part, this currently duplicates some code.

Comment (kohr-h) [reply odl/util/testutils.py]: Please check the latest version, not sure what you exactly meant here with the second part.

File: odl/discr/discr_mappings.py [hunk 7]
```diff
@@ -580,39 +581,52 @@ def __init__(self, fspace, partition, dspace, schemes, **kwargs):
...
-        try:
-            schemes_ = str(schemes + '').lower()  # pythonic string check
-            schemes_ = [schemes_] * self.grid.ndim
-        except TypeError:
-            schemes_ = [str(scm).lower() if scm is not None else None
-                        for scm in schemes]
+        schemes_in = schemes
+        if is_string(schemes):
+            schemes = [str(schemes).lower()] * self.grid.ndim
+            if schemes[0] not in _SUPPORTED_INTERP_SCHEMES:
```
Comment (adler-j) [inline odl/discr/discr_mappings.py]: Reorder this so you don't have to do `[0]`

File: odl/test/largescale/trafos/fourier_slow_test.py [hunk 8]
```diff
@@ -27,33 +27,20 @@
 # --- pytest fixtures --- #
 
 
-impl_params = [never_skip('numpy'), skip_if_no_pyfftw('pyfftw')]
-impl_ids = [" impl = '{}'".format(p.args[1]) for p in impl_params]
-
-
 # bug in pytest (ignores pytestmark) forces us to do this this
+impl_params = [never_skip('numpy'), skip_if_no_pyfftw('pyfftw')]
 largescale = " or not pytest.config.getoption('--largescale')"
```
Comment (adler-j) [inline odl/test/largescale/trafos/fourier_slow_test.py]: Don't we have a utility for this? E.g. `skip_if_no_largescale`?

Comment (kohr-h) [reply odl/test/largescale/trafos/fourier_slow_test.py]: We do, but how to use it here? I don't think I can wrap the `skipif` thing again in a `skipif`, at least not without getting ugly results in `ids`.

Comment (kohr-h) [reply odl/test/largescale/trafos/fourier_slow_test.py]: Just tested, wrapping again doesn't work.

**Comment by kohr-h** [comment]
Rebased and fixed the first comment. Merging after CI.

**Pull request merged**